### PR TITLE
Set BOOST_LANG_HIP to the HIP version number

### PR DESCRIPTION
**Pull request by fwyzard**
Set `BOOST_LANG_HIP` to the HIP version number, as returned by `BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, 0)`.

Note that HIP 4.5.0 has `HIP_VERSION_MINOR == 4`, so is identified as version 4.4.x.

### CODE REVIEW

**Comment by fwyzard** [comment]
This failure

> docs/readthedocs.org:alpaka — Read the Docs build failed! 

is likely unrelated to the changes in the PR...

**Review by bernhardmgruber — DISMISSED** [review]

**Review by j-stephan — DISMISSED** [review]

**Review by bernhardmgruber — APPROVED** [review]

File: include/alpaka/core/BoostPredef.hpp
Comment (psychocoderHPC) [inline include/alpaka/core/BoostPredef.hpp:22]: Could you please add `HIP_VERSION_PATCH` instead of `0` for the last number.

**Review by psychocoderHPC — CHANGES_REQUESTED** [review]

File: include/alpaka/core/BoostPredef.hpp
Comment (fwyzard) [reply include/alpaka/core/BoostPredef.hpp:22]: I thought about, but `HIP_VERSION_PATCH` is too large: `BOOST_VERSION_NUMBER` leaves four digits for the patch number, but _e.g._ in ROCm 5.0.x `HIP_VERSION_PATCH` is `13602` :-/

Comment (fwyzard) [reply include/alpaka/core/BoostPredef.hpp:22]: And it's the same in 5.0.0, 5.0.1 and 5.0.2, so it would not be useful to distinguish them, either.

**Comment by fwyzard** [comment]
@psychocoderHPC @j-stephan is there anything holding this up ?

**Comment by j-stephan** [comment]
Not from my side, I'm waiting for @psychocoderHPC's approval.

File: include/alpaka/core/BoostPredef.hpp
Comment (psychocoderHPC) [reply include/alpaka/core/BoostPredef.hpp:22]: ok makes sense to keep it in this case as `0`

**Review by psychocoderHPC — APPROVED** [review]

**Pull request merged**
